### PR TITLE
Guild awaitAvailiable method and removing redundant code in guild structure

### PR DESCRIPTION
**Pull request by ZiomaleQ**
Since, there was no way of checking if guild is "complete". I added this, kinda hacky sollution tbh.

### CODE REVIEW

**Review by Helloyunho — APPROVED** [review]
lgtm

**Comment by DjDeveloperr** [comment]
There's a new event now: guildLoaded which fires when a guild is loaded, user can await the event using Promise and conditional checks.

**Comment by DjDeveloperr** [comment]
Just a small UX improvement, what about resolving to loaded `Guild` from the method?

File: src/structures/guild.ts [hunk 1]
```diff
@@ -343,6 +305,22 @@ export class Guild extends Base {
       }
     })
   }
+
+  /**
+   * Fulfills promise when guild becomes available
+   */
+  async awaitAvailability(): Promise<Guild> {
+    return await new Promise((resolve, reject) => {
+      if(!this.unavailable) return;
```
Comment (DjDeveloperr) [inline src/structures/guild.ts:314]: Instead of returning, resolve to `this`.

File: src/structures/guild.ts [hunk 2]
```diff
@@ -343,6 +305,22 @@ export class Guild extends Base {
...
+      const listener = (guild: Guild): void => {
+        if (guild.id === this.id) {
+          this.client.removeListener('guildLoaded', listener);
+          resolve(this);
+        }
+      };
+      this.client.on('guildLoaded', listener);
```
Comment (DjDeveloperr) [inline src/structures/guild.ts:321]: If the event never occurs, this listener always stays... memory leaks can be there. So, add a timeout to reject and make it configurable through a param in the method.

**Review by DjDeveloperr — CHANGES_REQUESTED** [review]

**Review by DjDeveloperr — APPROVED** [review]
LGTM - thanks! 🚀

**Pull request merged**